Update modules.md (#3273)
cashapp/misk: docs/modules.md

Misk Modules
===============
Misk is split into many Gradle subprojects to organize functionality. The below is an overview of some of the modules.

In general, our module strategy is **TODO**

We use testfixtures **TODO*

## misk

Most of the implementation of Misk's web server components.

> There is a lot of stuff in this module. Pieces are occasionally extracted into new modules.

Reviewer: ```suggestion
This is the original module, more and more pieces are being extracted into new modules for improved organization, faster builds, and smaller dependencies for downstream customers.
```

fail the request without much boilerplate.

## misk-api

> High level interfaces and data classes. This module is agnostic to implementation details.

Reviewer: ```suggestion
High level interfaces and data classes which are implemented by both misk and wisp modules. This module is agnostic to implementation details.
```

## misk-events

An event publisher + consumer. There is no open source reference implementation at this time.

## wisp.*

These modules were created to extract specific pieces of functionality out of the `misk*` modules
into new, low-dependency modules. They were especially focused on having no Guice dependencies.

> Some of these modules duplicate existing Misk functionality.

Reviewer: ```suggestion
Some of these modules duplicate existing Misk functionality, but over time implementations will be deduplicated as part of broader code cleanup efforts.
```

> Now, a few of these modules are starting to be folded back into `misk*`, or at least deduplicated.

Reviewer: ```suggestion
```

This might be a bit forward for public docs.
